src/test/java/org/reaktivity/nukleus/mqtt/internal/ConnectionIT.java

```diff
@@ -970,8 +1029,24 @@ public void shouldRejectWillRetainWithoutWillFlag() throws Exception
     @Configure(name = WILDCARD_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
     @Configure(name = SHARED_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
     @Configure(name = MAXIMUM_QOS_NAME, value = "2")
+    @Configure(name = SESSION_EXPIRY_INTERVAL_NAME, value = "0")
     public void shouldRejectSubscribeWithNoLocal() throws Exception
     {
         k3po.finish();
     }
+
+    @Test
+    @Specification({
+        "${routeExt}/session/server/controller",
+        "${client}/connect/session.with.will.message/client",
+        "${server}/publish.session.data/server"})
+    @Configure(name = CLIENT_ID_NAME, value = "one")
+    @Configure(name = WILDCARD_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = SHARED_SUBSCRIPTION_AVAILABLE_NAME, value = "true")
+    @Configure(name = MAXIMUM_QOS_NAME, value = "2")
+    @Configure(name = SESSION_EXPIRY_INTERVAL_NAME, value = "0")
+    public void shouldPublishWillMessageAfterClientDisconnect() throws Exception
```

Comment:
We need two variations of this test, one where the client disconnects abruptly (no `DISCONNECT` packet), which sends the `will message` to the `will topic`, and on where the client disconnects normally (with `DISCONNECT` packet) ending the session, which does not send the `will message`, even though it was specified in the `CONNECT`.